modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -1733,9 +1732,25 @@ void ONNXImporter::handleNode(const opencv_onnx::NodeProto& node_proto_)
             if (!hasVariableInps)
             {
                 std::vector<Mat> inputs(node_proto.input_size()), concatenated;
+                // Due constant folding we can get inputs with different number of dimensions
+                // Insert the missing dimension to inputs
+                MatShape inputShape;
                 for (size_t i = 0; i < inputs.size(); ++i)
                 {
                     inputs[i] = getBlob(node_proto, i);
+                    if (inputs[i].size.dims() > inputShape.size())
+                    {
+                        inputShape = shape(inputs[i]);
+                    }
+                }
+
+                int axis = layerParams.get<int>("axis", 1);
```

Comment:
It make sense to add comment with link with description of default axis value (modern opsets doesn't have default): https://github.com/onnx/onnx/blob/master/docs/Changelog.md#Concat-1